### PR TITLE
Improvements to TVsCE 02

### LINKED ISSUE #1017: [Feature]: Implement contract registry in TVsCE
### Describe the feature you would like

- [ ] Add contract to registry
- [ ] Remove contract from registry
- [ ] A visual distinction for contracts that are in the registry

### Is this feature solving a problem?

Details about the problem your feature solves

### Please share any additional details you have?

Details about the problem your feature solves

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #1018: Implement test related commands in TVsCE
### What happened?

Separate commands for test registration and running

### What should have happened?

Details about what should have happened.

### Steps to reproduce the bug

Steps to reproduce the error

### Relevant log output

_No response_

### Screenshots and images

Screenshot and images

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #1022: [Bug]: The GUI compile command can compile the wrong contract
### What happened?

Reported by @alexzbusko 

I have access to set of files that caused this behaviour, will try to reproduce

### What should have happened?

Details about what should have happened.

### Steps to reproduce the bug

Steps to reproduce the error

### Relevant log output

_No response_

### Screenshots and images

Screenshot and images

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by AlirezaHaghshenas**
closes #1017 
closes #1022 
closes #1018

### CODE REVIEW

**Review by mweichert — APPROVED** [review]

File: taqueria-vscode-extension/src/lib/helpers.ts [hunk 1]
```diff
@@ -740,6 +762,52 @@ export class VsCodeHelper {
 		);
 	}
 
+	exposeTestSetupCommand(projectDir?: Util.PathToDir) {
```
Comment (jchenche) [inline taqueria-vscode-extension/src/lib/helpers.ts:765]: The parameter `projectDir` is optional but the body of this function uses `projectDir` without checking if it's provided.

File: taqueria-vscode-extension/src/lib/helpers.ts [hunk 2]
```diff
@@ -740,6 +762,52 @@ export class VsCodeHelper {
...
+						progressTitle: `Setting up folder ${folder} as test`,
+					},
+					projectDir,
+					false,
+				);
+			},
+		);
+	}
+
+	exposeRunTestCommand(projectDir?: Util.PathToDir) {
```
Comment (jchenche) [inline taqueria-vscode-extension/src/lib/helpers.ts:793]: The parameter `projectDir` is optional but the body of this function uses `projectDir` without checking if it's provided.

File: taqueria-vscode-extension/src/lib/helpers.ts [hunk 3]
```diff
@@ -912,6 +985,7 @@ export class VsCodeHelper {
 	private dataProviders: { refresh: () => void }[] = [];
 	private contractsDataProvider?: ContractsDataProvider;
 	private sandboxesDataProvider?: SandboxesDataProvider;
+	private testDataProvider?: TestDataProvider;
```
Comment (jchenche) [inline taqueria-vscode-extension/src/lib/helpers.ts:988]: Why do we make this field optional? Don't we always assign a value to it in the constructor?

File: taqueria-vscode-extension/src/lib/helpers.ts (hunk 1 above)
Comment (AlirezaHaghshenas) [reply taqueria-vscode-extension/src/lib/helpers.ts:776]: Good point. Thank Jimmy.

File: taqueria-vscode-extension/src/lib/helpers.ts (hunk 3 above)
Comment (AlirezaHaghshenas) [reply taqueria-vscode-extension/src/lib/helpers.ts:999]: Not in the constructor, but when the method `registerDataProviders` is called. If we make it required, static analysis will complain that it has no value when the class is instantiated (either direct value when defining field or in constructor).

**Review by jchenche — APPROVED** [review]

**Pull request merged**